java/core/src/java/org/apache/orc/impl/RunLengthIntegerWriterV2.java

```diff
@@ -755,7 +755,8 @@ public void write(long val) throws IOException {
 
           // if fixed runs reached max repeat length then write values
           if (fixedRunLength == MAX_SCOPE) {
-            determineEncoding();
+            encoding = EncodingType.DELTA;
+            isFixedDelta = true;
```

Comment:
Ditto. This PR makes line 763 a little obsolete.
> // variable delta run

Reply:
I think the `else` on line 762 separates the fixed delta run from the variable delta run. 763 This is followed by the variable delta run.

Reply:
However, java and C++ have different styles of processing
```java
//Java
if (prevDelta == 0 && currentDelta == 0) {
  // fixed delta run
} else {
  // variable delta run
}
```

```C++
//c++
if (prevDelta == 0 && currentDelta == 0) {
  // fixed delta run
  return;
} 
// variable delta run
```